includes/wikia/Wikia.php

```diff
@@ -2248,6 +2249,30 @@ static function onImportHandlePageXMLTagFilter ( WikiImporter $importer, &$pageI
 	}
 
 	/**
+	 * Restrict editinterface right to whitelist
+	 * set $result true to allow, false to deny, leave alone means don't care
+	 * usually return true to allow processing other hooks
+	 * return false stops permissions processing and we are totally decided (nothing later can override)
+	 */
+	static function canEditInterfaceWhitelist (&$title, &$wgUser, $action, &$result) {
+		global $wgEditInterfaceWhitelist;
+
+		// List the conditions we don't care about for early exit
+		if ( $action == "read" || $title->getNamespace() != NS_MEDIAWIKI || empty( $wgEditInterfaceWhitelist )) {
+			return true;
+		}
+
+		// In this NS, editinterface applies only to white listed pages and staff users
+		if (in_array($title->getDBKey(), $wgEditInterfaceWhitelist) || in_array('staff', $wgUser->getGroups())) {
```

Comment:
What do you think about making `$wgEditInterfaceWhitelist` case-insensitive and using `mb_lowercase()` in the first part of the condition?